rpc/src/p2p.rs

```diff
@@ -1,17 +1,66 @@
+use celestia_types::p2p::{
+    AddrInfo, BandwidthStats, Connectedness, PeerId, Reachability, ResourceManagerStats,
+};
 use jsonrpsee::proc_macros::rpc;
-use serde::{Deserialize, Serialize};
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct AddrInfo {
-    #[serde(rename = "ID")]
-    pub id: String,
-    // TODO: multiaddr
-    #[serde(rename = "Addrs")]
-    pub addrs: Vec<String>,
-}
 
 #[rpc(client)]
 pub trait P2P {
+    #[method(name = "p2p.BandwidthForPeer")]
+    async fn p2p_bandwidth_for_peer(&self, peer_id: &PeerId) -> Result<BandwidthStats, Error>;
+
+    #[method(name = "p2p.BandwidthForProtocol")]
+    async fn p2p_bandwidth_for_protocol(&self, protocol_id: &str) -> Result<BandwidthStats, Error>;
+
+    #[method(name = "p2p.BandwidthStats")]
+    async fn p2p_bandwidth_stats(&self) -> Result<BandwidthStats, Error>;
+
+    /// This method does not report errors due to a workaround to a go-jsonrpc bug, see https://github.com/eigerco/celestia-node-rs/issues/53
+    #[method(name = "p2p.BlockPeer")]
+    async fn p2p_block_peer(&self, peer_id: &PeerId);
+
+    /// This method does not report errors due to a workaround to a go-jsonrpc bug, see https://github.com/eigerco/celestia-node-rs/issues/53
+    #[method(name = "p2p.ClosePeer")]
+    async fn p2p_close_peer(&self, peer_id: &PeerId);
+
+    /// This method does not report errors due to a workaround to a go-jsonrpc bug, see https://github.com/eigerco/celestia-node-rs/issues/53
+    #[method(name = "p2p.Connect")]
+    async fn p2p_connect(&self, address: &AddrInfo);
+
+    #[method(name = "p2p.Connectedness")]
+    async fn p2p_connectedness(&self, peer_id: &PeerId) -> Result<Connectedness, Error>;
+
     #[method(name = "p2p.Info")]
     async fn p2p_info(&self) -> Result<AddrInfo, Error>;
+
+    #[method(name = "p2p.IsProtected")]
+    async fn p2p_is_protected(&self, peer_id: &PeerId, tag: &str) -> Result<bool, Error>;
+
+    #[method(name = "p2p.ListBlockedPeers")]
+    async fn p2p_list_blocked_peers(&self) -> Result<Vec<PeerId>, Error>;
+
+    #[method(name = "p2p.NATStatus")]
+    async fn p2p_nat_status(&self) -> Result<Reachability, Error>;
+
+    #[method(name = "p2p.PeerInfo")]
+    async fn p2p_peer_info(&self, peer_id: &PeerId) -> Result<AddrInfo, Error>;
+
+    #[method(name = "p2p.Peers")]
+    async fn p2p_peers(&self) -> Result<Vec<PeerId>, Error>;
+
+    /// This method does not report errors due to a workaround to a go-jsonrpc bug, see https://github.com/eigerco/celestia-node-rs/issues/53
+    #[method(name = "p2p.Protect")]
+    async fn p2p_protect(&self, peer_id: &PeerId, tag: &str);
+
+    #[method(name = "p2p.PubSubPeers")]
+    async fn p2p_pub_sub_peers(&self, topic: &str) -> Result<Option<Vec<PeerId>>, Error>;
```

Comment:
In other places that a list was returned you had `Vec<PeerId>` instead of `Option<Vec<PeerId>>`. Remove `Option` to keep it consistent.

Reply:
This might be another artifact of `go-jsonrpc`, but for topics without subs (or nonsense topics we send in tests) we get `null` in response rather than empty list. This causes deserialisation to fail, if it's not wrapped in `Option`. 
For other calls (peer list, block list, etc.) we get empty list as expected.